### PR TITLE
feat: add RIP-7560 related APIs

### PR DESCRIPTION
**Pull request by kangsorang**
# Description

Add RIP-7560 related APIs

- EstimateRip7560TransactionGas

- CallRip7560Validation

### CODE REVIEW

File: internal/ethapi/rip7560api.go [hunk 1]
```diff
@@ -41,6 +57,190 @@ func (s *TransactionAPI) GetRip7560TransactionDebugInfo(hash common.Hash) (map[s
...
+
+	// TODO(sm-stack): Configure RIP-7560 enabled devnet option
+	//header, err := headerByNumberOrHash(ctx, s.b, *blockNrOrHash)
+	//if err != nil {
+	//	return nil, err
+	//}
+
+	//if s.b.ChainConfig().IsRIP7560(header.Number) {
+	//	return nil, fmt.Errorf("cannot call RIP-7560 validation on pre-rip7560 block %v", header.Number)
+	//}
```
Comment (sm-stack) [inline internal/ethapi/rip7560api.go:74]: Not for local devnets, but we can now uncomment this for Pioneer Alpha, right?

File: eth/gasestimator/gasestimator_rip7560.go [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+}
+
+func EstimateRip7560Validation(ctx context.Context, tx *types.Transaction, opts *Options, gasCap uint64) (uint64, error) {
+	// Binary search the gas limit, as it may need to be higher than the amount used
+	st := tx.Rip7560TransactionData()
+	totalGasLimit, err := st.TotalGasLimit()
+	if err != nil {
+		return 0, err
+	}
+	gasLimit := totalGasLimit - st.PostOpGas
```
Comment (sm-stack) [inline eth/gasestimator/gasestimator_rip7560.go]: Maybe it won't affect the result, but shouldn't it be like this?
```suggestion
	gasLimit := totalGasLimit - st.Gas - st.PostOpGas
```

File: core/state_processor_rip7560.go [hunk 3]
```diff
@@ -352,7 +352,13 @@ func ApplyRip7560ValidationPhases(
 	header *types.Header,
 	tx *types.Transaction,
 	cfg vm.Config,
+	estimateFlag ...bool,
```
Comment (sm-stack) [inline core/state_processor_rip7560.go]: I'm not sure, but how do you think about making the name of this flag `allowSigFail`, as in at `abiDecodeAcceptAccount`?

File: internal/ethapi/transaction_args.go [hunk 4]
```diff
@@ -537,9 +582,9 @@ func (args *TransactionArgs) ToTransaction() *types.Transaction {
 			Sender:                      args.Sender,
 			AuthorizationData:           *args.AuthorizationData,
 			Paymaster:                   args.Paymaster,
-			PaymasterData:               *args.PaymasterData,
+			PaymasterData:               toByte(args.PaymasterData),
```
Comment (sm-stack) [inline internal/ethapi/transaction_args.go:585]: Just curious, what was the intention of this?

Comment (kangsorang) [reply internal/ethapi/transaction_args.go:585]: Code to prevent nil pointer reference from occurring

**Review by sm-stack — APPROVED** [review]

**Pull request merged**